plugins/BEdita/Core/src/ORM/Inheritance/AssociationCollection.php

```diff
@@ -56,11 +57,34 @@ public function __construct(Table $table)
         // Copy existing associations to the new collection. This is the new collection. Associations are copied here.
         $this->_items = $table->associations()->_items;
 
+        $this->innerCollection = new CakeAssociationCollection();
         if ($table->inheritedTable() !== null) {
             $this->innerCollection = $table->inheritedTable()->associations();
         }
     }
 
+    /**
+     * Get the inherited associations collection.
+     *
+     * The clone of `self::innerCollection` is cleaned by `RelatedTo` associations
+     * that involve concrete objects.
+     *
+     * @return \Cake\ORM\AssociationCollection
+     */
+    protected function inheritedAssociations()
+    {
+        $innerCollection = clone $this->innerCollection;
```

Comment:
Mindfuck question: what happens if `$this->innerCollection` is itself a BEdita `AssociationCollection`? It is cloned, but `$this->innerCollection->innerCollection` and `$innerCollection->innerCollection` still point to the same instance. 😱 So if an association removal cascades, it is actually removed from the actual AssociationCollection, right? 🤔(Don't ask me to read it again, because I wouldn't understand what I wrote myself.)

Maybe we should add something like this:
```php
public function __clone()
{
    $this->innerCollection = clone $this->innerCollection;
}
```

Reply:
:scream: I think you are right! I'll try your approach and make a unit test on it.